web/frontend/angular/src/app/main-pages/register/register.component.html

```diff
@@ -5,12 +5,30 @@ <h2 class="register-header">Register</h2>
     class="register-container"
     (ngSubmit)="register()"
   >
-    <p [ngClass]="{'has-error': isSubmitted && formControls.email.errors}">
-      <input type="email" placeholder="Email" formControlName="email" />
+    <div class="row d-none" id="error">
+      <span>Invalid Credentials</span>
```

Comment:
You need to be dynamically changing the error message to correspond to the actual error that occurred. "Invalid credentials" is not always an appropriate error message.

Here is an example error that DRF may return
```json
{"username":["A user with that username already exists."]}
```

We use the default error configuration for DRF, so you can look at the documentation to get an understanding of the error response formats https://www.django-rest-framework.org/api-guide/exceptions/ Notice that validation errors (like the above example) are handled differently by DRF.

That being said, not _all_ errors will follow this format, because sometimes there may be an internal server error (500+ status code). In that case just display "internal error" as the message or something like that.

Reply:
Not every time Api response is like this. On postman i am getting response like this. But on Angular iam getting errror like this."HttpErrorResponse {headers: HttpHeaders, status: 400, statusText: 'Bad Request', url: 'http://0.0.0.0:8000/api/accounts/register/', ok: false, …}"

Reply:
According to documentation for [`HttpErrorResponse`](https://angular.io/api/common/http/HttpErrorResponse#description=),

> The error property will contain either a wrapped Error object or the error response returned from the server.

So, check the `error` property of that response.